internal/driver/metadata.go

```diff
@@ -173,22 +179,25 @@ func getStreamingParameters(d *usbdevice.Device) (interface{}, error) {
 	if err != nil {
 		return nil, err
 	}
-	result := StreamingParameters{}
+	streamingParameters := StreamingParameters{}
 
 	tpf := DescNotSpecified
 	if sp.Capture.Capability == v4l2.StreamParamTimePerFrame {
 		tpf = DescTimePerFrame
 	}
-	result.Capability = StreamingCapability{tpf, sp.Capture.Capability}
+	streamingParameters.Capability = StreamingCapability{tpf, sp.Capture.Capability}
 
 	hiqual := DescNotSpecified
 	if sp.Capture.CaptureMode == v4l2.StreamParamModeHighQuality {
 		hiqual = DescHighQuality
 	}
-	result.CaptureMode = CaptureMode{hiqual, sp.Capture.CaptureMode}
+	streamingParameters.CaptureMode = CaptureMode{hiqual, sp.Capture.CaptureMode}
+
+	streamingParameters.TimePerFrame = sp.Capture.TimePerFrame
+	streamingParameters.ReadBuffers = sp.Capture.ReadBuffers
 
-	result.TimePerFrame = sp.Capture.TimePerFrame
-	result.ReadBuffers = sp.Capture.ReadBuffers
+	result := make(map[string]StreamingParameters)
+	result[d.Name()] = streamingParameters
```

Comment:
what is this doing? formatting it as the API would return it?

Reply:
It's adding the path name to the data structure so that when the value is returned, its made clear that the information is specifically for that video path
so it looks like this: 
```
 "objectValue": {
      "/dev/video6": {
          "ImageFormats": [
              {
                  "BufType": 1,
                  "Description": "YUYV 4:2:2",
                  "Flags": 0,
                  "FrameSizes": [
                      {
                          "Index": 0,
                          "PixelFormat": 1448695129,
                          "Size": {
                              "MaxHeight": 
```

Reply:
Along with this there is also `Path` field in the output? Are these both not the same? Also do we really need to format like this, if so then should the same be done for other command outputs?

Reply:
It is done for all of the applicable outputs. Not every metadata command changes based on path.